packages/ui/src/FileCard/index.tsx

```diff
@@ -0,0 +1,170 @@
+      );
+    }
+    if (onDelete) {
+      actionButtons.push(
+        <Button icon="pi pi-trash" onClick={() => onDelete?.(file)} />,
+      );
+    }
+    if (onDownload) {
+      actionButtons.push(
+        <Button icon="pi pi-download" onClick={() => onDownload?.(file)} />,
+      );
+    }
+    if (onShare) {
+      actionButtons.push(
+        <Button icon="pi pi-share-alt" onClick={() => onShare?.(file)} />,
+      );
+    }
+    if (onView) {
+      actionButtons.push(
+        <Button icon="pi pi-eye" onClick={() => onView?.(file)} />,
+      );
+    }
+
+    return actionButtons;
+  };
+
+  return (
+    <Card className="file-card">
+      <div className="file-thumbnail-details-wrapper">
+        {renderThumbnailType()}
+        <div className="file-details-wrapper">
+          <div className="file-name-description-details-wrapper">
+            <div>
+              <span className="file-name">{file.filename}</span>
+              <span className="file-size">
+                {file.size && showSize && `(${file?.size})`}
+              </span>
+            </div>
+            {showDescription && file.description && (
+              <>
+                <div className="file-description-details">
+                  <span>{file.description}</span>
+                  {showEditDescription && (
+                    <Button icon="pi pi-pencil" text size="small" />
+                  )}
+                </div>
+              </>
+            )}
+          </div>
+          <div className="file-upload-download-details-wrapper">
+            <div className="file-upload-details">
+              <div className="uploaded-by">
+                <span>{messages?.uploadedbyHeader || "Uploaded by"}</span>
+                <span>{checkUploadedByData(file)}</span>
+              </div>
+              <div className="uploaded-at">
+                <span>{messages?.uploadedAtHeader || "Uploaded at"}</span>
+                <span>{updateDateFormat(file?.uploadedAt)}</span>
+              </div>
+            </div>
+            <div className="file-download-details">
+              <div className="download-count">
+                {(file?.downloadCount || file?.downloadCount === 0) && (
+                  <>
+                    <span>{messages?.downloadCountLabel || "Downloads:"}</span>
+                    <span>{file?.downloadCount}</span>
+                  </>
+                )}
+              </div>
+              <div className="last-downloaded-at">
+                {file.lastDownloadedAt && (
+                  <>
+                    <span>
+                      {messages?.lastDownloadedAtHeader || "Last download:"}
+                    </span>
+                    <span>{updateDateFormat(file.lastDownloadedAt)}</span>
+                  </>
+                )}
+              </div>
+            </div>
+          </div>
+          <div className="file-actions">{...renderActions()}</div>
```

Comment:
this can be moved inside renderActions. renderActions should return the whole block including this div, not just the buttons array.
```suggestion
         {renderActions()}
```